File: FitnessClub/ClientApp/src/app/models/subscription.ts
import { Discount } from "./discount"
import { Order } from "./order"
import { Review } from "./review"
import { Wishlist } from "./wishlist"

export interface Subscription {
  id: number,
  pricePerDay: number,
  title: NonNullable<string>,
  rating: number,

  discounts: Array<Discount>,
  orders: Array<Order>,
  reviews: Array<Review>,
  wishlists: Array<Wishlist>
}

File: FitnessClub/Models/Subscription.cs
using System;
using System.Collections.Generic;

namespace FitnessClub.Models;

public partial class Subscription
{
    public int Id { get; set; }

    public decimal PricePerDay { get; set; }

    public string Title { get; set; } = null!;

    public decimal? Rating { get; set; }

    public virtual ICollection<Discount> Discounts { get; } = new List<Discount>();

    public virtual ICollection<Order> Orders { get; } = new List<Order>();

    public virtual ICollection<Review> Reviews { get; } = new List<Review>();

    public virtual ICollection<Wishlist> Wishlists { get; } = new List<Wishlist>();
}
